prog/mutation.go

```diff
@@ -370,33 +371,205 @@ func mutationArgs(c *Call) (args, bases []*Arg, parents []*[]*Arg) {
 	return
 }
 
+func swap16(v uint16) uint16 {
+	v0 := byte(v >> 0)
+	v1 := byte(v >> 8)
+	v = 0
+	v |= uint16(v1) << 0
+	v |= uint16(v0) << 8
+	return v
+}
+
+func swap32(v uint32) uint32 {
+	v0 := byte(v >> 0)
+	v1 := byte(v >> 8)
+	v2 := byte(v >> 16)
+	v3 := byte(v >> 24)
+	v = 0
+	v |= uint32(v3) << 0
+	v |= uint32(v2) << 8
+	v |= uint32(v1) << 16
+	v |= uint32(v0) << 24
+	return v
+}
+
+func swap64(v uint64) uint64 {
+	v0 := byte(v >> 0)
+	v1 := byte(v >> 8)
+	v2 := byte(v >> 16)
+	v3 := byte(v >> 24)
+	v4 := byte(v >> 32)
+	v5 := byte(v >> 40)
+	v6 := byte(v >> 48)
+	v7 := byte(v >> 56)
+	v = 0
+	v |= uint64(v7) << 0
+	v |= uint64(v6) << 8
+	v |= uint64(v5) << 16
+	v |= uint64(v4) << 24
+	v |= uint64(v3) << 32
+	v |= uint64(v2) << 40
+	v |= uint64(v1) << 48
+	v |= uint64(v0) << 56
+	return v
+}
+
 func mutateData(r *randGen, data []byte) []byte {
 	for stop := false; !stop; stop = r.bin() {
 		r.choose(
-			1, func() {
+			100, func() {
+				// Append byte.
 				data = append(data, byte(r.rand(256)))
 			},
-			1, func() {
+			100, func() {
+				// Remove byte.
+				if len(data) == 0 {
+					return
+				}
+				i := r.Intn(len(data))
+				copy(data[i:], data[i+1:])
+				data = data[:len(data)-1]
+			},
+			100, func() {
+				// Replace byte with random value.
 				if len(data) == 0 {
 					return
 				}
 				data[r.Intn(len(data))] = byte(r.rand(256))
 			},
-			1, func() {
+			100, func() {
+				// Flip bit in byte.
 				if len(data) == 0 {
 					return
 				}
 				byt := r.Intn(len(data))
 				bit := r.Intn(8)
 				data[byt] ^= 1 << uint(bit)
 			},
-			1, func() {
+			100, func() {
+				// Swap two bytes.
+				if len(data) < 2 {
+					return
+				}
+				i1 := r.Intn(len(data))
+				i2 := r.Intn(len(data))
+				data[i1], data[i2] = data[i2], data[i1]
+			},
+			100, func() {
+				// Add / subtract from a byte.
 				if len(data) == 0 {
 					return
 				}
 				i := r.Intn(len(data))
-				copy(data[i:], data[i+1:])
-				data = data[:len(data)-1]
+				delta := byte(r.rand(35) + 1)
+				if r.rand(2) == 0 {
+					data[i] += delta
+				} else {
+					data[i] -= delta
+				}
+			},
+			100, func() {
+				// Add / subtract from a uint16.
+				if len(data) < 2 {
+					return
+				}
+				i := r.Intn(len(data) - 1)
+				delta := uint16(r.rand(35) + 1)
+				switch r.rand(4) {
+				case 0:
+					*(*uint16)(unsafe.Pointer(&data[i])) += delta
+				case 1:
+					*(*uint16)(unsafe.Pointer(&data[i])) -= delta
+				case 2:
+					x := *(*uint16)(unsafe.Pointer(&data[i]))
+					*(*uint16)(unsafe.Pointer(&data[i])) = swap16(swap16(x) + delta)
+				case 3:
+					x := *(*uint16)(unsafe.Pointer(&data[i]))
+					*(*uint16)(unsafe.Pointer(&data[i])) = swap16(swap16(x) - delta)
+				}
+			},
+			100, func() {
+				// Add / subtract from a uint32.
+				if len(data) < 4 {
+					return
+				}
+				i := r.Intn(len(data) - 3)
+				delta := uint32(r.rand(35) + 1)
```

Comment:
Few simplifications possible.
1. Declare the magic const 35 at the top of the function as:
	const maxInc = 35
2. unsafe can be used only once to get a typed pointer.
3. make delta [-maxInc, maxInc] to halve number of cases.
4. use r.bin
As the result  this can be reduced to:

```go
	p := (*uint32)(unsafe.Pointer(&data[i]))
	delta := uint32(r.rand(2 * maxInc + 1) - maxInc)
	if r.bin() {
		*p += delta
	} else {
		*p = swap32(swap32(*p) + delta)
	}
```

The same applies to uint8/uint16/uin64.
If we make delta [-maxInc, maxInc], then we will add 0 with 1/70-th probability, which is a no-op. We can leave it as is, or add: if delta == 0 { delta = 1 }